src/graph_ir/low_graph_ir.h

```diff
@@ -0,0 +1,60 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+/*
+ * This is a small low-level IR that contains particles with flattened 
+ * input/output access path lists. It can be translated directly into datalog
+ * relations that similarly represent particles with flattened lists of full 
+ * access path names.
+ *
+ * There is likely a higher-level IR from which this compiles, and the higher 
+ * level IR can be generated by deserializing protobufs.
+ */
+
+
+#include <string>
+#include <vector>
+#include <set>
+
+#ifndef LOW_GRAPH_IR_
+#define LOW_GRAPH_IR_
+
+using namespace std;
+
+class LowGraphAccessPath {
+    public:
+        string full_path_name;
```

Comment:
In general, you should make fields private and mediate access to them through methods.

Reply:
done